modules/sdk-coin-atom/src/atom.ts

```diff
@@ -226,6 +257,223 @@ export class Atom extends BaseCoin {
+        amount: [{ denom: 'uatom', amount: GAS_AMOUNT }],
+        gasLimit: GAS_LIMIT,
+      };
+      const actualBalance = Number(balance) - Number(gasBudget.amount[0].amount);
+      const amount: Coin[] = [
+        {
+          denom: 'uatom',
+          amount: actualBalance.toString(),
+        },
+      ];
+
+      const sendMessage: SendMessage[] = [
+        {
+          fromAddress: senderAddress,
+          toAddress: params.recoveryDestination,
+          amount: amount,
+        },
+      ];
+
+      // Step 7: Build the unsigned tx
+      const txnBuilder = this.getBuilder().getTransferBuilder();
+      txnBuilder
+        .messages(sendMessage)
+        .gasBudget(gasBudget)
+        .publicKey(publicKey)
+        .accountNumber(Number(accountNumber))
+        .chainId(chainId);
+      const unsignedTransaction = (await txnBuilder.build()) as Transaction;
+      let serializedTx = unsignedTransaction.toBroadcastFormat();
+      const signableHex = unsignedTransaction.signablePayload.toString('hex');
+
+      // Step 8: Validate necessary parameters required for performing recovery
+      if (!params.userKey) {
+        throw new Error('missing userKey');
+      }
+      if (!params.backupKey) {
+        throw new Error('missing backupKey');
+      }
+      if (!params.walletPassphrase) {
+        throw new Error('missing wallet passphrase');
+      }
+
+      const userKey = params.userKey.replace(/\s/g, '');
+      const backupKey = params.backupKey.replace(/\s/g, '');
+      const [userKeyCombined, backupKeyCombined] = ((): [
+        ECDSAMethodTypes.KeyCombined | undefined,
+        ECDSAMethodTypes.KeyCombined | undefined
+      ] => {
+        const [userKeyCombined, backupKeyCombined] = this.getKeyCombinedFromTssKeyShares(
+          userKey,
+          backupKey,
+          params.walletPassphrase
+        );
+        return [userKeyCombined, backupKeyCombined];
+      })();
+
+      if (!userKeyCombined || !backupKeyCombined) {
+        throw new Error('Missing key combined shares for user or backup');
+      }
+
+      // Step 9: Sign the tx
+      const signature = await this.signRecoveryTSS(userKeyCombined, backupKeyCombined, signableHex);
+      const atomKeyPair = new AtomKeyPair({ pub: publicKey });
+      txnBuilder.addSignature({ pub: atomKeyPair.getKeys().pub }, Buffer.from(signature.r + signature.s, 'hex'));
+      const signedTransaction = await txnBuilder.build();
+      serializedTx = signedTransaction.toBroadcastFormat();
+
+      return { serializedTx: serializedTx, scanIndex: i };
+    }
+    throw new Error('Did not find an address with funds to recover');
+  }
+
+  /**
+   * Get balance from public node
+   */
+  protected async getBalanceFromNode(senderAddress: string): Promise<request.Response> {
+    const nodeUrl = this.getPublicNodeUrl();
+    const getBalancePath = '/cosmos/bank/v1beta1/balances/';
+    const fullEndpoint = nodeUrl + getBalancePath + senderAddress;
```

Comment:
I think, fullnode URL ending with forward slash and getBalancePath is starting with forward slash, will it create an issue ?

Reply:
good catch, yeah that will be a problem; will edit